### PR TITLE
Port redmine issues 831 to 937 to the new testsuite

### PR DESCRIPTION
**Pull request by D4N**
- The decorators which copy & delete files now call the `setUp` & `tearDown` functions in a more logical order (`setUp` is called after the copy is made and `tearDown` before the file is deleted).
- I have modified the actual test for issue `937a`. The original test would run the following:
```
    runTest exiv2 -pa    $filename | sort            > $num-before.txt
    runTest exiv2 -m     $dataname  $filename
    runTest exiv2 -pa    $filename | sort            > $num-after.txt
    diff $diffargs $num-before.txt $num-after.txt    > $num.txt
    diff $diffargs $num.txt        $diffname
```
However, the first command produces no output at all, thus the diff `$num.txt` is just `$num-after.txt` prepended with `> `. To be honest, I then don't really get the point of comparing that diff to another diff (if I understood the redmine discussion correctly, the original diff is the output before the respective patch got applied). For simply testing the behavior of exiv2, it should be sufficient to just check the output of the second invocation of `exiv2 -pa $filename`.

### CODE REVIEW

**Comment by D4N** [comment]
@clanmills Could you take a look at the test for issue 836? It is failing on Travis on OSX (the only platform that has resource forks), but I am unsure if it is a regression or my mistake. Also, please note that resource forks are now accessible via `$filename/..namedfork/rsrc` and not `$filename/rsrc`.

**Comment by D4N** [comment]
> @clanmills Could you take a look at the test for issue 836? It is failing on Travis on OSX (the only platform that has resource forks), but I am unsure if it is a regression or my mistake. Also, please note that resource forks are now accessible via $filename/..namedfork/rsrc and not $filename/rsrc.

Nevermind, I forgot to copy the resource fork correctly. It seems to work now.

File: tests/bugfixes/redmine/test_issue_841.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+from system_tests import CaseMeta, path
+
+
+class CrashOnInput(metaclass=CaseMeta):
```
Comment (piponazo) [inline tests/bugfixes/redmine/test_issue_841.py]: [suggestion] Should we rename the test from `crash...` to `throw...` ? The application was crashing before the fix, but now it is throwing an exception.

File: tests/bugfixes/redmine/test_issue_855.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+from system_tests import CaseMeta, path
+
+
+class SegfaultWhenAccessingFocalLength(metaclass=CaseMeta):
```
Comment (piponazo) [inline tests/bugfixes/redmine/test_issue_855.py]: [suggestion] Same comment here. I guess the application was crashing before, but now we are controlling the situation.

File: tests/bugfixes/redmine/test_issue_884.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    stdout = [
+        """Exif.Pentax.LensType                         Byte        2  Tamron SP AF 17-50mm F2.8 XR Di II
+""",
+        """Exif.Pentax.LensType                         Byte        2  smc PENTAX-F 35-70mm F3.5-4.5
+""",
+        """Exif.Pentax.LensType                         Byte        2  PENTAX-F 28-80mm F3.5-4.5
+"""
+    ]
+    stderr = [""] * 3
```
Comment (piponazo) [inline tests/bugfixes/redmine/test_issue_884.py]: [suggestion] What about removing the hard-coded value (`3`) and use `[""] * len(commands)` ?

**Review by piponazo — APPROVED** [review]
I just made few suggestions, feel free to take actions about them or not. Good job!

File: tests/bugfixes/redmine/test_issue_841.py (hunk 1 above)
Comment (D4N) [reply tests/bugfixes/redmine/test_issue_841.py]: Uh, yeah, now that I re-read the class name, it sounds like it should crash...

**Pull request merged**